### PR TITLE
add ut for policy priority

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
Signed-off-by: changzhen <changzhen5@huawei.com>

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Add ut for policy priority.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/detector/compare.go [hunk 1]
```diff
@@ -1,9 +1,62 @@
...
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/util"
+	"github.com/karmada-io/karmada/pkg/util/fedinformer/keys"
+)
+
+func getHighestPriorityPropagationPolicies(policies []*policyv1alpha1.PropagationPolicy, resource *unstructured.Unstructured, objectKey keys.ClusterWideKey) *policyv1alpha1.PropagationPolicy {
+	var (
+		matchedPolicy         *policyv1alpha1.PropagationPolicy
+		matchedPolicyPriority = util.PriorityMisMatch
+	)
```
Comment (RainbowMango) [inline pkg/detector/compare.go:16]: We barely use the variable block.

File: pkg/detector/compare.go [hunk 2]
```diff
@@ -1,9 +1,62 @@
...
+	}
+	klog.V(2).Infof("Matched policy(%s/%s) for resource(%s)", matchedPolicy.Namespace, matchedPolicy.Name, objectKey)
+	return matchedPolicy
+}
+
+func getHighestPriorityClusterPropagationPolicies(policies []*policyv1alpha1.ClusterPropagationPolicy, resource *unstructured.Unstructured, objectKey keys.ClusterWideKey) *policyv1alpha1.ClusterPropagationPolicy {
+	var (
+		matchedClusterPolicy         *policyv1alpha1.ClusterPropagationPolicy
+		matchedClusterPolicyPriority = util.PriorityMisMatch
+	)
```
Comment (RainbowMango) [inline pkg/detector/compare.go:39]: ditto.

File: pkg/detector/compare_test.go [hunk 3]
```diff
@@ -78,12 +81,20 @@ func Test_GetHigherPriorityPropagationPolicy(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "Test 4",
```
Comment (RainbowMango) [inline pkg/detector/compare_test.go:85]: We'd better give each case a human-readable name so that we can go through and check the coverage easier.

Comment (XiShanYongYe-Chang) [reply pkg/detector/compare_test.go]: Updated.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**